### PR TITLE
Migrate FW-190D9 to Module

### PR DESCRIPTION
**Pull request by jdahlblom**

### CODE REVIEW

File: Scripts/DCS-BIOS/lib/modules/aircraft_modules/FW-190D9.lua [hunk 1]
```diff
@@ -0,0 +1,224 @@
+module("FW-190D9", package.seeall)
+
+local Module = require("Module")
+
+--- @class FW_190A8: Module
+local FW_190A8 = Module:new("FW-190D9", 0x3400, { "FW-190D9" })
```
Comment (charliefoxtwo) [inline Scripts/DCS-BIOS/lib/modules/aircraft_modules/FW-190D9.lua:6]: Variable should probably be named FW_190D9?

File: Scripts/DCS-BIOS/lib/modules/aircraft_modules/FW-190D9.lua [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+FW_190A8:definePotentiometer("SIGHT_WING_SPAN", 21, 3002, 131, { 0, 1 }, "Gun Sight", "EZ42 Gunsight Target Wingspan Knob")
+FW_190A8:defineToggleSwitch("SIGHT_NIGHT", 21, 3011, 134, "Gun Sight", "EZ42 Gunsight Night Lens Filter, ON/OFF")
+FW_190A8:defineRotary("SIGHT_DISC_CTRL", 21, 3003, 4, "Gun Sight", "EZ42 Gunsight Target Distance Control")
+FW_190A8:defineRotary("SIGHT_BRIGHT", 21, 3004, 135, "Gun Sight", "EZ42 Gunsight Brightness Control")
+FW_190A8:defineRotary("SIGHT_ALT_KNOB", 21, 3012, 191, "Gun Sight", "EZ42 Gunsight Altitude Aiming Correction Adjusting Knob I")
+FW_190A8:defineRotary("INSTR_LGHT_CTRL", 7, 3001, 78, "Cockpit", "Instrument Lights Brightness")
+FW_190A8:defineRotary("FUG16_VOLUME", 15, 3003, 83, "Radio", "Radio Volume")
+FW_190A8:defineToggleSwitch("FT_ZF_SWITCH", 15, 3006, 84, "Radio", "FT FT / Y ZF Mode Switch")
+-- Allocate 16 bits of address space to avoid changing the address of subsequent allocations:
+FW_190A8:allocateInt(65535)
```
Comment (charliefoxtwo) [inline Scripts/DCS-BIOS/lib/modules/aircraft_modules/FW-190D9.lua:51]: This function shouldn't be public I don't think. Module:reserveIntValue is public and does the same thing - it's just a bit clearer what it's doing.

File: Scripts/DCS-BIOS/lib/modules/aircraft_modules/FW-190D9.lua [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+FW_190A8:defineFloat("ADF_HORIZONT", 30, { -1, 1 }, "Indicator", "ADF_Horizont")
+FW_190A8:defineFloat("WINDSHIELDDAMAGE", 209, { 0, 1 }, "Indicator", "WindshieldDamage")
+FW_190A8:defineFloat("CANOPYDAMAGE", 210, { 0, 1 }, "Indicator", "CanopyDamage")
+
+FW_190A8:defineFixedStepTumb("FUG16_TUNING", 15, 3008, 82, 0.01, { -1, 1 }, { -0.01, 0.01 }, nil, "Radio", "Radio Tuning (+/-30 kHz)")
+FW_190A8:defineTumb("STARTER_SWITCH2", 2, 3009, 105, 0.5, { 0, 1 }, nil, false, "Cockpit", "Starter Switch. Magnetic Clutch")
+FW_190A8:define3PosTumb("FUEL_COLD_START", 6, 3007, 160, "Cockpit", "Cold Start/Wind Screen Washer")
+
+--Externals
+FW_190A8:defineIntegerFromGetter("EXT_POSITION_LIGHT_LEFT", function()
```
Comment (charliefoxtwo) [inline Scripts/DCS-BIOS/lib/modules/aircraft_modules/FW-190D9.lua:175]: Module:defineBitFromDrawArgument can be used to clean up these external calls

File: Scripts/DCS-BIOS/lib/modules/aircraft_modules/FW-190D9.lua (hunk 1 above)
Comment (jdahlblom) [reply Scripts/DCS-BIOS/lib/modules/aircraft_modules/FW-190D9.lua:6]: of course

Comment (jdahlblom) [reply Scripts/DCS-BIOS/lib/modules/aircraft_modules/FW-190D9.lua:6]: fixed

File: Scripts/DCS-BIOS/lib/modules/aircraft_modules/FW-190D9.lua (hunk 3 above)
Comment (jdahlblom) [reply Scripts/DCS-BIOS/lib/modules/aircraft_modules/FW-190D9.lua]: done

**Comment by jdahlblom** [comment]
json did not change

File: Scripts/DCS-BIOS/BIOS.lua [hunk 4]
```diff
@@ -80,8 +80,15 @@ dofile(lfs.writedir()..[[Scripts/DCS-BIOS/lib/FC3.lua]]) -- ID = 4, ProperName =
 -- dofile(lfs.writedir()..[[Scripts/DCS-BIOS/lib/FW-190A8.lua]]) -- ID = 21, ProperName = Fw 190 A-8 Anton
 local FW_190A8 = require "FW-190A8"
 BIOS.protocol.writeNewModule(FW_190A8)
-dofile(lfs.writedir()..[[Scripts/DCS-BIOS/lib/FW-190D9.lua]]) -- ID = 22, ProperName = Fw 190 D-9 Dora
+
+--dofile(lfs.writedir()..[[Scripts/DCS-BIOS/lib/FW-190D9.lua]]) -- ID = 22, ProperName = Fw 190 D-9 Dora
+local FW_190D9 = require "FW-190D9"
+BIOS.protocol.writeNewModule(FW_190D9)
+
+-- dofile(lfs.writedir()..[[Scripts\DCS-BIOS\lib\I-16.lua]]) -- ID = 23, ProperName = I-16
```
Comment (charliefoxtwo) [inline Scripts/DCS-BIOS/BIOS.lua:88]: Looks like a conflict resolution artifact? I don't think this should be here. Same with the extra new lines added below

Comment (jdahlblom) [reply Scripts/DCS-BIOS/BIOS.lua:87]: fixed

**Pull request merged**